step01/main.py

```diff
@@ -92,113 +54,104 @@ def create_thumbnail_key(key):
 
 def create_image_record(bucket, key):
   '''
-  creates an Image object to store in DB
-  stores in Images table
+    Creates an Image object to store in DB.
+    Record stored in Images table.
 
-  :param bucket: S3 bucket to use
-  :param key: path to image in specified S3 bucket
-  :return: Images Object
+    :param bucket: S3 bucket to use
+    :param key: path to image in specified S3 bucket
+    :returns: Images Object
   '''
   return Images(bucket=bucket, filename=PurePath(key).name,
                 key=key)
 
 
 def main():
+  # setting up SQSQueue to process SQSMessage
+  sqs_message = SQSQueue(QUEUE_URL, client=sqs)
   print('Waiting for Messages...')
+
   while True:
-    response = sqs.receive_message(
-      QueueUrl=QUEUE_URL,
-      MaxNumberOfMessages=1,
-      WaitTimeSeconds=2
-    )
-
-    messages = gen_messages_from_response(response)
-    for message in messages:
-      try:
-        get_object_response = s3.get_object(
-          Bucket=BUCKET_NAME,
-          Key=message.key
-        )
-
-        print(f'Creating Thumbnail for {message._id}')
-        stream = get_object_response['Body'].read()
-        thumbnail_stream = create_thumbnail(stream)
-
-        print(f'Uploading Thumbnail for {message._id}')
-        thumbnail_key = create_thumbnail_key(message.key)
-
-        # creates and adds image record to DB
-        image_for_database = create_image_record(BUCKET_NAME, message.key)
-        db.session.add(image_for_database)
-
-        # updates image record with thumbnail key
-        image_for_database.thumbnail_key = thumbnail_key
-        db.session.commit()
-
-        s3.put_object(
-          Bucket=BUCKET_NAME,
-          Key=thumbnail_key,
-          Body=thumbnail_stream
-        )
-
-        print(f'Deleting Message {message._id}')
-        sqs.delete_message(
-          QueueUrl=QUEUE_URL,
-          ReceiptHandle=message.receipt_handle
-        )
-
-      except ClientError as boto_error:
-        '''        
+    try:
```

Comment:
I haven't really mentioned this before, but it's typically better to put inside the `try` block small amounts of code you're trying to catch exceptions for. So rather than:
```python
try:
  foo()
  bar()
except FooError:
  pass
except BarError:
  pass
```

It's more readable to do the following because it's easier to tell what causes what exception:
```python
try:
  foo()
except FooException:
  pass

try:
  bar()
except BarException:
  pass
```

Can you rearrange the code so that all the exceptions are caught individually? There can still be a try catch block to keep the while loop from exiting if that's the intended design.